packages/core/src/CorePlugin.ts

```diff
@@ -130,41 +129,32 @@ export class CorePlugin extends JWPlugin {
         }
         this.editor.vDocument.deleteSelection();
     }
-    /**
-     * Navigate to a given range.
-     *
-     * @param params
-     */
-    setSelection(params: VSelectionParams): void {
-        this.editor.vDocument.selection.set(params.vSelection);
-        // Each time the selection changes, we reset its format.
-        this.editor.vDocument.formatCache = null;
-    }
 
     /**
-     * Insert 4 spaces or indent multiples lines by adding 4 spaces in the beginning of the lines.
+     * Indent text or lines.
      *
-     * - If the range is collapsed, insert 4 spaces
-     * - If there is more than one line selected in range, indent each lines
-     * - If there is only one line selected in range, insert 4 space
+     * - If there is more than one line selected in range, indent each lines.
+     * - Otherwise, insert 4 spaces.
      */
     indent(params: IndentParams): void {
         const range = params.range || this.editor.vDocument.selection.range;
-        if (range.isCollapsed()) {
-            this.editor.vDocument.insertText(params.range, this.indentChars);
-        } else {
-            const firstCharsOfLines = this.getFirstForEachLineInRange(range);
-            // If there is at least two lines
-            if (firstCharsOfLines.length > 1) {
-                // For each `firstCharsOfLines`, add all the `indentChars` in the document
-                firstCharsOfLines.forEach((firstCharNode: VNode) => {
-                    this.editor.vDocument.withRange(VRange.at(firstCharNode), range => {
-                        this.editor.vDocument.insertText(range, this.indentChars);
-                    });
+        const isSegmentBreak = (node: VNode): boolean => {
```

Comment:
The following lines are reapeated in both function:
```
const isSegmentBreak = (node: VNode): boolean => {
const numberOfSpacePerIndent = this.indentChars.length;	            return node instanceof LineBreakNode || isChar(node.firstChild());
};
const segmentBreaks = range.selectedNodes.filter(isSegmentBreak);
const tabLength = this.indentChars.length;
// Only outdent when there is at leat two lines selected	        // Only outdent when there is at leat two lines selected
if (firstChars.length > 1) {	        if (segmentBreaks.length) {
    firstChars.forEach((firstCharVNode: VNode) => {	            // The first line is usually only partially selected, which means
        const spaceChars = [];	            // the segment break preceding it was not in `range.selectedNodes`.
        let currentCharVNode = firstCharVNode;	            segmentBreaks.unshift(range.start.previous(isSegmentBreak));
```

Is the function `getFirstForEachLineInRange` containing that logic more flexible?